api/librdb-ext-api.h

```diff
@@ -158,15 +155,29 @@ _LIBRDB_API RdbxRespFileWriter *RDBX_createRespFileWriter(RdbParser *p,
 /****************************************************************
  * Create RESP to Redis TCP connection
  *
- * If provided path is NULL then write stdout
  * Can configure pipeline depth of transmitted RESP commands. Set
  * to 0 if to use default.
  ****************************************************************/
-_LIBRDB_API RdbxRespToTcpLoader *RDBX_createRespToTcpLoader(RdbParser *p,
+typedef enum RdbxRedisConnType {
+    RDBX_CONN_TCP,
+    RDBX_CONN_USER_FD,
+} RdbxRedisConnType;
+
+typedef struct RdbxRedisConn {
+    RdbxRedisConnType type;
+    int pipelineDepth;
+    union {
+        int fd;                         /* file descriptor */
+        struct {
+            const char *hostname;
+            int port;
+        } tcp;                          /* TCP connection */
+    } u;
+} RdbxRedisConn;
+
+_LIBRDB_API RdbxRespToRedisLoader *RDBX_createRespToRedis(RdbParser *p,
                                                             RdbxToResp *rdbToResp,
-                                                            const char* hostname,
-                                                            int port,
-                                                            int pipelineDepth);
+                                                            RdbxRedisConn *redisConn);
```

Comment:
i don't think taking a struct pointer, let alone one with union, is a good interface.
maybe we should expose several "create" functions?
and if they have many common attributes, we can move these to some "set"ter functions?

Reply:
this API is suppose to be externally available, right?
with stable ABI and multi-language binding...

Reply:
Ok, I will split it to `RDBX_createRespToRedisTcp` and `RDBX_createRespToRedisFd`